lib/fs/casefs.go

```diff
@@ -370,77 +377,87 @@ func (r *defaultRealCaser) realCase(name string) (string, error) {
 		return out, nil
 	}
 
-	r.mut.Lock()
-	defer r.mut.Unlock()
-
-	node := r.root
+	var node *caseNode
 	for _, comp := range strings.Split(name, string(PathSeparator)) {
-		if node.dirNames == nil || node.expires.Before(time.Now()) {
+		// Check cache
+		v, ok := r.cache.Get(out)
+		if ok {
+			node = v.(*caseNode)
+			if node.expires.Before(time.Now()) {
+				r.purgeRecursive(out, node.children)
```

Comment:
Technically we only need to purge recursively if the casing actually changed at this level, otherwise we can let child cache entries live on their own... Not sure if that makes anything easier or harder.

Reply:
We can also ignore the recursive purge entirely... Children entries could then outlive their parent, but only by at most one cache time, so an entry that lives for one second might refer to parent information up to two seconds old... I think that would be fine?

Reply:
Thinking further, this might not even be a problem since the cache is keyed on "out" (real case)? The children would become unreachable since we wouldn't find them by traversal any more, and get expelled from the cache at some point.